### PR TITLE
Feat/set status for amendments and revisions

### PR DESCRIPTION
**Pull request by Sepehr-Sobhani**
[ZH Card](https://app.zenhub.com/workspaces/climate-action-secretariat-60ca4121764d710011481ca2/issues/bcgov/cas-cif/981)

### CODE REVIEW

File: app/cypress/integration/cif/project-revision-change-logs.spec.js
Comment (tmastrom) [inline app/cypress/integration/cif/project-revision-change-logs.spec.js:44]: The card calls this "Approved" but I believe it was decided that this would be renamed to "Applied" throughout?

**Comment by Sepehr-Sobhani** [comment]
@pbastia based on your comment here: https://app.zenhub.com/workspaces/climate-action-secretariat-60ca4121764d710011481ca2/issues/bcgov/cas-cif/1166
Do we need to commit the revision once the status has changed to `Applied/Approved` ?! if yes, do I need to add it to the AC of this card or make a new card for that?!

File: app/pages/cif/project-revision/[projectRevision]/view.tsx
Comment (pbastia) [inline app/pages/cif/project-revision/[projectRevision]/view.tsx:113]: This widget should be in its own file

File: app/pages/cif/project-revision/[projectRevision]/view.tsx
Comment (pbastia) [inline app/pages/cif/project-revision/[projectRevision]/view.tsx:210]: no need to iterate over the statuses if the revision type is "Amendment", something like that might work better:

```suggestion
  const filteredRevisionStatuses = 
    projectRevision.revisionType === "Amendment" ? 
        allRevisionStatuses.edges :
        allRevisionStatuses.edges.filter(({ node }) => !node.isAmendmentSpecific );
```

File: app/pages/cif/project-revision/[projectRevision]/view.tsx
Comment (pbastia) [inline app/pages/cif/project-revision/[projectRevision]/view.tsx:27]: the two parameters are inconsistent (we should type them to avoid it and increase readability)
- `allRevisionTypes` is an object `{ edges: { node: ... }[]}`
- `allRevisionStatuses` is an array of edges

I suggest making them either both an array of edges, and calling them `allRevisionTypesEdges` and `allRevisionStatusesEdges`

File: app/pages/cif/project-revision/[projectRevision]/view.tsx
Comment (pbastia) [inline app/pages/cif/project-revision/[projectRevision]/view.tsx:229]: casting here could be avoided if we declare the return type of `createProjectRevisionViewSchema(..)`

**Review by pbastia — CHANGES_REQUESTED** [review]
Good stuff!!

File: app/pages/cif/project-revision/[projectRevision]/view.tsx
Comment (pbastia) [inline app/pages/cif/project-revision/[projectRevision]/view.tsx:33]: - the `readonly` is unnecessary here
- we can remove the edges key to make both arguments consistent in type

```suggestion
const createProjectRevisionViewSchema = (
  allRevisionTypesEdges: {
     node: {
       type: string;
      };
  }[],
  allRevisionStatusesEdges: {
   node: {
     name: string;
     isAmendmentSpecific: boolean;
    };
  }[]
```

**Review by pbastia — CHANGES_REQUESTED** [review]
I probably just confused you a little with my comment earlier :p

**Review by pbastia — APPROVED** [review]
🥇

**Pull request merged**
